### PR TITLE
Feature/db provides existing parameters values

### PR DESCRIPTION
**Pull request by so7fie**
I added methods in BookDatabase and SaleDatabase to provide the set of possible values of the different query parameters (ig. SaleOrdering values, BookCondition values). I also implemented a `DummyInterestDatabase` that provide list of interests that would exists in our regular database. 

This pr is useful for the filtering activities, because they need to display all the possible choices of parameter values to the user
## Pull reviewers stats
Stats for the last 180 days:
|                                                                                                                                                                 | User            | Total reviews | Median time to review                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Total comments |
| --------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------- | ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------- |
| <a href="https://github.com/Huntsekkern"><img src="https://avatars.githubusercontent.com/u/44240415?v=4" width="20"></a>                                        | Huntsekkern     | 13            | [51m](https://app.flowwer.dev/charts/review-time/~(u~(i~'44240415~n~'Huntsekkern)~p~180~r~(~(d~'qpumtn~t~'1745)~(d~'qpt0zy~t~'2wmx)~(d~'qpt88n~t~'33vm)~(d~'qptd1d~t~'29r)~(d~'qptd3y~t~'2cc)~(d~'qqjij5~t~'i6m)~(d~'qq8faa~t~'qv)~(d~'qq8fiz~t~'bv)~(d~'qpt4tp~t~'2b1)~(d~'qpuo7b~t~'mk)~(d~'qq6uip~t~'3j5)~(d~'qrihqg~t~'1wf3)~(d~'qrm9wa~t~'17x)~(d~'qrma4k~t~'1g7)~(d~'qrmajv~t~'1vi)~(d~'qrni3f~t~'15)~(d~'qqkomy~t~'z8i)~(d~'qqkozj~t~'zl3)~(d~'qqkp4h~t~'zq1)~(d~'qqvo8r~t~'14d)~(d~'qrmb5j~t~'1cc)~(d~'qrs0pv~t~'3yyp)~(d~'qrvgff~t~'1lvl))))                                       | **31**         |
| <a href="https://github.com/so7fie"><img src="https://avatars.githubusercontent.com/u/10992808?v=4" width="20"></a>                                             | so7fie          | 13            | [19h 10m](https://app.flowwer.dev/charts/review-time/~(u~(i~'10992808~n~'so7fie)~p~180~r~(~(d~'qpuy0b~t~'1v)~(d~'qqew7j~t~'2pkp)~(d~'qqf92s~t~'32fy)~(d~'qqjbho~t~'1ux)~(d~'qq0t0v~t~'5m7q)~(d~'qq0t2c~t~'5m97)~(d~'qq0p5z~t~'5hby)~(d~'qqy5l4~t~'5im1)~(d~'qq7qp0~t~'hf)~(d~'qrm6v4~t~'6ai)~(d~'qqkh75~t~'dr0)~(d~'qqrud7~t~'7qx2)~(d~'qr8kh3~t~'1kfr)~(d~'qrm9jc~t~'1npk)~(d~'qrmbez~t~'1ls)~(d~'qrpf7u~t~'11ut)~(d~'qrr5wv~t~'1e1c)~(d~'qrnshx~t~'193))))                                                                                                                                | 24             |
| <a href="https://github.com/RaulinN"><img src="https://avatars.githubusercontent.com/u/44499915?u=c029696ff2de0902f905fbf35188359e94e08d88&v=4" width="20"></a> | RaulinN         | **15**        | [3h 48m](https://app.flowwer.dev/charts/review-time/~(u~(i~'44499915~n~'RaulinN)~p~180~r~(~(d~'qptcqf~t~'1yt)~(d~'qpq5rb~t~'5xjz)~(d~'qqr6wy~t~'aju)~(d~'qreyev~t~'f)~(d~'qqhz4w~t~'87bj)~(d~'qq73x3~t~'8mkh)~(d~'qq821x~t~'lj5)~(d~'qq73ig~t~'jy6)~(d~'qq7vmj~t~'1t5)~(d~'qpuwak~t~'8pt)~(d~'qq73am~t~'28)~(d~'qrni37~t~'x)~(d~'qq7qbn~t~'1bz)~(d~'qqj0lo~t~'u5)~(d~'qqjmtu~t~'n2b)~(d~'qrm5d5~t~'2qe)~(d~'qrrzpp~t~'3mco)~(d~'qrsjsf~t~'7tp)~(d~'qrryuq~t~'428q)~(d~'qrrz93~t~'42n3)~(d~'qrs0cu~t~'3ylo))))                                                                               | 24             |
| <a href="https://github.com/Snowflake8"><img src="https://avatars.githubusercontent.com/u/44499990?v=4" width="20"></a>                                         | Snowflake8      | 14            | [3h 8m](https://app.flowwer.dev/charts/review-time/~(u~(i~'44499990~n~'Snowflake8)~p~180~r~(~(d~'qpt3s9~t~'2zf8)~(d~'qqjlvf~t~'b)~(d~'qqxzb3~t~'ikt)~(d~'qqdm38~t~'1fge)~(d~'qq6ncv~t~'8609)~(d~'qq8el9~t~'y2h)~(d~'qqjh1g~t~'gox)~(d~'qqjh87~t~'7lg)~(d~'qpx29n~t~'1vgi)~(d~'qq0q7g~t~'5jeb)~(d~'qq6k9y~t~'8hk)~(d~'qq6kh9~t~'8ov)~(d~'qq6r5v~t~'6b)~(d~'qqsnpq~t~'qn)~(d~'qrm4bb~t~'1ljt)~(d~'qrnhv0~t~'7)~(d~'qrkbo5~t~'88t)~(d~'qrm4eh~t~'1rq)~(d~'qro2s6~t~'666))))                                                                                                                    | 20             |
| <a href="https://github.com/DGoedtkindtEPFL"><img src="https://avatars.githubusercontent.com/u/61052800?v=4" width="20"></a>                                    | DGoedtkindtEPFL | 14            | [19h 49m](https://app.flowwer.dev/charts/review-time/~(u~(i~'61052800~n~'DGoedtkindtEPFL)~p~180~r~(~(d~'qpuy16~t~'2q)~(d~'qpuoy1~t~'co)~(d~'qpup0b~t~'19at)~(d~'qplma9~t~'1e2x)~(d~'qpuugt~t~'am9h)~(d~'qqrtyv~t~'xlr)~(d~'qqjmgc~t~'dy)~(d~'qqjjxz~t~'9s4m)~(d~'qq9yj2~t~'1i5c)~(d~'qq6s0w~t~'8gm)~(d~'qq6rfm~t~'fn8)~(d~'qqxoq1~t~'21ln)~(d~'qqioj1~t~'2x8a)~(d~'qqionj~t~'2xcs)~(d~'qqioou~t~'2xe3)~(d~'qqipcq~t~'2y1z)~(d~'qqru2n~t~'7qmi)~(d~'qrm4pz~t~'1iw7)~(d~'qrm4za~t~'1j5i)~(d~'qrm5eq~t~'1jky)~(d~'qrmfv4~t~'1u1c)~(d~'qrmhyq~t~'1cp)~(d~'qrnxgy~t~'1wd)~(d~'qrsrti~t~'4q2c)))) | 8              |
| <a href="https://github.com/Laniakea1999"><img src="https://avatars.githubusercontent.com/u/43450242?v=4" width="20"></a>                                       | Laniakea1999    | 5             | [**32m**](https://app.flowwer.dev/charts/review-time/~(u~(i~'43450242~n~'Laniakea1999)~p~180~r~(~(d~'qpupsq~t~'d4)~(d~'qqeyk1~t~'2rx7)~(d~'qq7oa5~t~'1n)~(d~'qrmbac~t~'1h5)~(d~'qrx7lz~t~'1024))))                                                                                                                                                                                                                                                                                                                                                                                          | 3              |

### CODE REVIEW

**Review by DGoedtkindtEPFL — CHANGES_REQUESTED** [review]
For the enums bookOrdering, bookConditions, etc... 
there is no need for methods in the databases; One can use the Enum to get the list of values. Also, I find it's out of place in the interface of the database.

You should also place the interface for the interest database in the interface folder.

**Comment by so7fie** [comment]
> For the enums bookOrdering, bookConditions, etc...
> there is no need for methods in the databases; One can use the Enum to get the list of values. Also, I find it's out of place in the interface of the database.
> 
> You should also place the interface for the interest database in the interface folder.

ok done :+1:

File: app/src/main/java/com/github/polybooks/utils/BookUtil.kt [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
+import com.github.polybooks.core.Course
+import com.github.polybooks.core.Field
+import com.github.polybooks.core.Semester
 
 fun anonymousBook(title: String): Book =
-        Book("", null, title, null, null, null, null, null)
\ No newline at end of file
+        Book("", null, title, null, null, null, null, null)
+
+val mockFields : List<Field> = listOf(
```
Comment (DGoedtkindtEPFL) [inline app/src/main/java/com/github/polybooks/utils/BookUtil.kt:11]: Il y aurait moyen de mettre ça dans la DummyInterestDatabase

**Review by Laniakea1999 — APPROVED** [review]
Nothing to say, I think it's good.

File: app/src/main/java/com/github/polybooks/utils/BookUtil.kt (hunk 1 above)
Comment (so7fie) [reply app/src/main/java/com/github/polybooks/utils/BookUtil.kt:11]: Courld we merge it as it is and fix it when we will do the build-variant thing?

Comment (so7fie) [reply app/src/main/java/com/github/polybooks/utils/BookUtil.kt]: okay its done

**Review by DGoedtkindtEPFL — APPROVED** [review]
Good for me now

**Pull request merged**